src/main/scala/com/fulcrumgenomics/bam/api/SamSource.scala

```diff
@@ -66,6 +65,29 @@ object SamSource {
     index.foreach(i => input.index(i))
     new SamSource(fac.open(input))
   }
+
+  /** Constructs a [[SamSource]] to read from the provided input stream.
+    *
+    * @param stream the input stream of SAM/BAM/CRAM bytes
+    * @param ref an optional reference sequencing for decoding CRAM files
+    * @param async if true use extra thread(s) to speed up reading
+    * @param stringency the validation stringency to apply when reading the data
+    * @param factory a SAMRecordFactory; MUST return classes that mix in [[SamRecord]]
+    */
+  def apply(
+    stream: InputStream,
+    ref: Option[PathToFasta],
+    async: Boolean,
+    stringency: ValidationStringency,
+    factory: SAMRecordFactory,
+  ): SamSource = {
+    val fac = SamReaderFactory.make()
+    fac.samRecordFactory(factory)
+    fac.setUseAsyncIo(async)
+    fac.validationStringency(stringency)
```

Comment:
do you want to refactor and have common code in the apply methods in one place?

Reply:
Yeah that could be nice, it would help future users from forgetting to set a parameter if a new entrypoint was added.